src/abstractdataframe/abstractdataframe.jl

```diff
@@ -197,14 +201,7 @@ eltypes(df)
 ```
 
 """
-function eltypes(df::AbstractDataFrame)
-    ncols = size(df, 2)
-    res = Array(Type, ncols)
-    for j in 1:ncols
-        res[j] = eltype(df[j])
-    end
-    return res
-end
+eltypes(df::AbstractDataFrame) = map!(eltype, Vector{Type}(size(df,2)), columns(df))
```

Comment:
So `Type[eltype(col) for col in columns(df)]` has element type `Any`, even with the type annotation in the front?